app-text/texlive/texlive-2021.recipe

```diff
@@ -659,66 +879,65 @@ INSTALL()
 	make texlinks
 
 	# and install the actual distribution...
-	cd ..
+	cd $sourceDir3
 	fetchTexLiveInstaller
-	# we should now be in "Master"
 	export TEXLIVE_INSTALL_PREFIX=$dataDir/texlive
 	export TEXLIVE_INSTALL_TEXMFSYSVAR=$sharedStateDir/texlive
 	export TEXLIVE_INSTALL_TEXMFSYSCONFIG=$sysconfDir/texlive
 	export TEXLIVE_INSTALL_TEXMFLOCAL=/boot/system/non-packaged/data/texmf-dist
 	export TEXLIVE_INSTALL_TEXMFVAR=/boot/home/config/var/texmf
 	export TEXLIVE_INSTALL_TEXMFCONFIG=/boot/home/config/settings/texmf/config
 	export TEXLIVE_INSTALL_TEXMFHOME=/boot/home/config/settings
-	if [ ! -e flag.patched ] ; then
-		patch -Np2 < $portDir/additional-files/tlutils.pm.patch
-		touch flag.patched
-	fi
 	# hmm, need to work-around /boot/system/non-packaged not existing....
 	# the texfm local tree should probably be done with a post-install script, perhaps
 	sed -i -e 's/make_local_skeleton \".*//' install-tl
 	# run the TeX Live installer, and specify all our paths, once more, to be sure... ;-)
-	export PATH=bin/i386-haiku:bin/x86_64-haiku:$prefix/bin:$PATH
-	install-tl <<< $"D
-1
-$dataDir/texlive
-2
-/boot/system/non-packaged/data/texlive/texmf-dist
-3
-$sharedStateDir/texlive
-4
-$sysconfDir/texlive
-5
-/boot/home/config/var/texmf
-6
-/boot/home/config/settings/texmf/config
-7
-/boot/home/config/settings/texmf-dist
-R
-I
-"
+	export PATH=bin/i386-haiku:bin/${targetArchitecture}-haiku:$binDir:$PATH
+	if [ $targetArchitecture = x86_gcc2 ] || [ $targetArchitecture = x86 ]; then
+		customBinDir=bin/i386-haiku/
+	else
+		customBinDir=bin/${targetArchitecture}-haiku/
+	fi
+	sed -e "s@dataDir@$dataDir@g" \
+		-e "s@sharedStateDir@$sharedStateDir@g" \
+		-e "s@sysconfDir@$sysconfDir@g" \
+		$portDir/additional-files/texlive.profile > texlive.profile
+	install-tl --custom-bin=$customBinDir --profile=texlive.profile || true
+
 	# somewhere along the way, a texmf.cnf gets created with ALL the wrong paths...
 	# replace it with the known working version
 	cp -f $dataDir/texmf-dist/web2c/texmf.cnf $dataDir/texlive/texmf-dist/web2c/texmf.cnf
 
 	# some reason, this doesn't get created, which we need
 	if [ ! -e $prefix/bin/mktexfmt ] ; then
-		ln -s $prefix/bin/fmtutil $prefix/bin/mktexfmt
+		ln -s fmtutil $prefix/bin/mktexfmt
 	fi
 
-	# remove the documents directory; it's too big for package create to handle!
-	# see https://dev.haiku-os.org/ticket/12116
-	#rm -rf $dataDir/texlive/texmf-dist/doc
+	# needed post-install steps apparently not done by install-tl
+	$binDir/fmtutil-sys --all
```

Comment:
This needs to be $prefix/bin instead of $binDir for 32bit:
```diff
- 	$binDir/fmtutil-sys --all
+ 	$prefix/bin/fmtutil-sys --all
```

Reply:
sorry, I missed that one